lib/pushmi_pullyu/aip/creator.rb

```diff
@@ -1,65 +1,40 @@
-require 'archive/tar/minitar'
+require 'minitar'
 require 'bagit'
-require 'pushmi_pullyu/aip'
+require 'fileutils'
 
 class PushmiPullyu::AIP::Creator
 
+  class BagInvalid < StandardError; end
+
   # Assumption: the AIP has already been downloaded
 
   def initialize(noid)
     @noid = noid
     @aip_directory = PushmiPullyu::AIP.aip_directory(noid)
     @aip_filename = PushmiPullyu::AIP.aip_filename(noid)
-    PushmiPullyu::AIP.validate(noid)
   end
 
-  def run(should_clean_work_directories: true)
+  def run
```

Comment:
Heavily simplify this. Creator class, should do just that, create the aip file (bagging and tarring). We don't need to do the cleanup of the directories or filename. We have a `PushmiPullyu::AIP.destroy(noid)` that is called that does exactly that. Also you don't need to check if the tar already exists and remove it. The tarring process will override an existing tar file if it already exists.